### PR TITLE
#20  Impelement ArrayMerge sniff in foreach

### PR DESCRIPTION
**Pull request by larsroettig**
Implementation for #20 

### Currently we have many findings in 2.3 develop

```

bin/phpcs --standard=Magento2 --sniffs=Magento2.Performance.ForeachArrayMerge  app/code app/lib
```

###  Magento 2 findings with current Implementation 
```
------------------------------------------------------------------------------------------------------
A TOTAL OF 0 ERRORS AND 280 WARNINGS WERE FOUND IN 212 FILES
------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by lenaorobei** [comment]
I checked some findings and they look false-positive to me. Probably, they require serious code refactoring, not just moving `array_merge` outside the loop.
Need more thoughts here to understand if we really want to warn developers about any case of usage  `array_merge` inside the loop.

File: Magento2/Sniffs/Performance/ForeachArrayMergeSniff.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
```
Comment (lenaorobei) [inline Magento2/Sniffs/Performance/ForeachArrayMergeSniff.php]:
```suggestion
```

Comment (lenaorobei) [inline Magento2/Sniffs/Performance/ForeachArrayMergeSniff.php]:
```suggestion
```

File: Magento2/Tests/Performance/ForeachArrayMergeUnitTest.inc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+}
+
+$selectBuilder = new SelectBuilder();
+
+foreach ([] as $collection) {
+    foreach ($configurationSources as $source) {
+        $selectBuilder->setColumns(array_merge($selectBuilder->getColumns(), $source));
+    }
+}
```
Comment (lenaorobei) [inline Magento2/Tests/Performance/ForeachArrayMergeUnitTest.inc]:
```suggestion
}

```

**Comment by buskamuza** [comment]
We should accept it, but include explanation of how to fix the problematic code. Think about a catalog of recommendations (either here in wiki or in devdocs).

**Comment by orlangur** [comment]
@larsroettig @buskamuza 

> A TOTAL OF 0 ERRORS AND 280 WARNINGS WERE FOUND IN 212 FILES

So how this rule can be useful?

**Pull request merged**